devtron-labs/devtron
# feat: show deployment status and timeline for helm apps deployed via gitops

# Description

This pr incorporates the feature for showing deployment status timelines for helm apps that are deployed via gitops. Currently we only show deployment status timeline for devtron apps that are deployed via gitops.

Fixes #2429 

## How Has This Been Tested?

- [x] I've tested this by deploying the changes in my cluster as well as locally. Following are the test scenarios that I've covered:- 
1. Check if entries for newly deployed helm charts are getting stored correctly in the respective tables, such as app, installed_apps, installed_app_versions, inatelled_app_version_history.
2. Check if the pipeline timeline status entries are being stored in these three tables or not for a corresponding helm chart:- pipeline_status_timeline, pipeline_status_timeline_resources and pipeline_status_timeline_sync_detail.
3. Check for multiple cases when the helm chart deployment status  is healthy , degraded, timed out. Progressing or etc.
4. Deploy a chart with random values, then its status should be degraded, then if the resources are in a progressing state for some time then the entry stored in the database should be Timed Out, check multiple cases like this.
5. Since this will also impact the code of all applications  deployed via gitops be it devtron also checks by deploying devton apps via gitops, it should work as before.
6. Check the behavior by simultaneously deploying multiple charts at once.
7. Status on previously deployed helm apps. 
8. Status on previously deployed devtron apps
9. Status on newly created devtron apps
10. Deploy charts in bulk and see if entries are getting stored correctly in db or not




## Checklist:

* [ ] Does this PR requires documentation updates?
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```
